04.ls/wc.rb

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env ruby
+
+# frozen_string_literal: true
+
+require 'optparse'
+
+def main
+  options, paths = receive_command_argument
+  paths.empty? ? output(count_stdin, options) : output(calculate_file_text(paths), options)
+end
+
+def receive_command_argument
+  opt_params = { l: false, w: false, c: false }
+  opt = OptionParser.new
+
+  opt.on('-l') { |v| opt_params[:l] = v }
+  opt.on('-w') { |v| opt_params[:w] = v }
+  opt.on('-c') { |v| opt_params[:c] = v }
+  argv = opt.parse(ARGV)
+  [opt_params, argv]
+end
+
+def count_stdin
+  inputs = $stdin.readlines.join
+  count = {
+    line: count_lines(inputs),
+    word: count_words(inputs),
+    byte: inputs.size
+  }
+  [count]
+end
+
+def calculate_file_text(paths)
+  paths.map do |file_name|
+    property = { line: 0, word: 0, byte: 0, name: '', directory: false }
+    if FileTest.file?(file_name)
+      file_texts = File.read(file_name)
+      property[:line] = count_lines(file_texts)
+      property[:word] = count_words(file_texts)
+      property[:byte] = file_texts.size
+    end
+    property[:name] = file_name
+    property[:directory] = FileTest.directory?(file_name)
+    property
+  end
+end
+
+def count_lines(text)
+  text.lines.size
+end
+
+def count_words(text)
+  text.split.size
+end
+
+def output(file_properties, options)
+  is_only_files = file_properties.none? { |property| property[:directory] == true }

```

Comment:
```suggestion
  is_only_files = file_properties.none? { |property| property[:directory] }
```

`== true`や`== false`は明示的に書く必要がないケースがほとんどです。
「プロを目指す人のためのRuby入門」の第2章のコラムにもそういう話が書いてあるので、もし持っていたら読んでみてください。

Reply:
修正し、チェックリストに追記しました